onnxruntime/core/framework/session_options.cc

```diff
@@ -57,4 +57,13 @@ Status SessionOptions::AddExternalInitializers(gsl::span<const std::string> name
   return Status::OK();
 }
 #endif  // !defined(ORT_MINIMAL_BUILD) && !defined(DISABLE_EXTERNAL_INITIALIZERS)
+
+void SessionOptions::AddCustomOpLibraryHandle(void* library_handle, const char* library_name) {
+  if (!custom_op_libs_) {
+    custom_op_libs_ = std::make_shared<std::vector<LibraryHandle>>();
+  }
+
+  custom_op_libs_->push_back(LibraryHandle(library_handle, library_name, Env::Default()));
```

Comment:
Just realized that this is still calling `Env::Default()` once per handle. So, we're not really gaining anything by passing in a reference to the env. I'll rethink.

Reply:
Env() thing is a small side issue. Since we are sharing handles as a whole between option instances, there is not a need to wrap them individually.